slsim/Deflectors/cluster_catalog_lens.py

```diff
@@ -0,0 +1,281 @@
+        members["center_x"] = (members["ra"] - center_ra) * 3600
+        members["center_y"] = (members["dec"] - center_dec) * 3600
+        center_dist = np.sqrt(members["center_x"] ** 2 + members["center_y"] ** 2)
+        members = members[center_dist < max_dist_arcsec]
+        return members
+
+    @staticmethod
+    def assign_similar_galaxy(
+        members_list,
+        galaxy_list,
+        cosmo=None,
+        bands=("g", "r", "i", "z", "Y"),
+        max_gals=10000,
+    ):
+        """Assigns a similar galaxy to each member of a group/cluster member catalog by
+        comparing their magnitudes and redshifts.
+
+        :param members_list: astropy table with columns 'mag_{band}', 'z'
+        :type members_list: astropy.table.Table
+        :param galaxy_list: astropy table with columns 'mag_{band}', 'z'
+        :type galaxy_list: astropy.table.Table
+        :param cosmo: astropy.cosmology instance
+        :type cosmo: astropy.cosmology
+        :param bands: list of bands to compare
+        :type bands: list
+        :param max_gals: maximum number of galaxies to compare to
+        :type max_gals: int
+        """
+        # shuffle galaxy list and select a subset
+        if len(galaxy_list) > max_gals:
+            indices = np.random.choice(len(galaxy_list), max_gals, replace=False)
+            galaxy_list = galaxy_list[indices]
+
+        mag_cols = [f"mag_{b}" for b in bands if f"mag_{b}" in members_list.columns]
+        if not bands:
+            raise ValueError("No magnitude columns found in members_list")
+        mag_members = [members_list[mag] for mag in mag_cols]
+        mag_galaxies = [galaxy_list[mag] for mag in mag_cols]
+        dist_mod_members = -5 * np.log10(
+            cosmo.luminosity_distance(members_list["z"]) / (10 * u.pc)
+        )
+        dist_mod_galaxies = -5 * np.log10(
+            cosmo.luminosity_distance(galaxy_list["z"]) / (10 * u.pc)
+        )
+        distance = cdist(
+            np.stack([*mag_members, dist_mod_members], axis=1),
+            np.stack([*mag_galaxies, dist_mod_galaxies], axis=1),
+            metric="euclidean",
+        )
+        nearest_neighbors_indices = distance.argmin(axis=1)
+        similar_galaxies = galaxy_list[nearest_neighbors_indices]
+        include_cols = [
+            col for col in members_list.columns if col not in mag_cols + ["z"]
+        ]
+        return hstack([members_list[include_cols], similar_galaxies])
+
+    def set_cosmo(self):
+        params = dict(
+            flat=(self.cosmo.Ok0 == 0.0),
+            H0=self.cosmo.H0.value,
+            Om0=self.cosmo.Om0,
+            Ode0=self.cosmo.Ode0,
+            Ob0=self.cosmo.Ob0 if self.cosmo.Ob0 is not None else 0.04897,
+            Tcmb0=self.cosmo.Tcmb0.value if self.cosmo.Tcmb0.value > 0 else 2.7255,
+            Neff=self.cosmo.Neff,
+            sigma8=0.8102,
+            ns=0.9660499,
+        )
+        colossus_cosmo.setCosmology(cosmo_name="halo_cosmo", **params)
```

Comment:
Just curious, is it necessary to use colossus cosmology or can be replaced by astropy?